src/rpc/misc.cpp

```diff
@@ -37,6 +40,219 @@
+            in.pushKV("vout", (int64_t)txin.prevout.n);
+
+            // Add address and value info if spentindex enabled
+            CSpentIndexValue spentInfo;
+            CSpentIndexKey spentKey(txin.prevout.hash, txin.prevout.n);
+            if (GetSpentIndex(spentKey, spentInfo)) {
+                if (tx.IsInvite()) {
+                    //invites are not demoninated in sotoshi.
+                    in.pushKV("amount", spentInfo.satoshis);
+                } else {
+                    in.pushKV("amount", ValueFromAmount(spentInfo.satoshis));
+                    in.pushKV("amountMicros", spentInfo.satoshis);
+                }
+
+                auto address = CMeritAddress{(signed char)(spentInfo.addressType), spentInfo.addressHash}.ToString();
+
+                in.pushKV("address", address);
+                isSender = isSender || address == walletAddress;
+
+                const auto maybe_referral = prefviewcache->GetReferral(spentInfo.addressHash);
+                if (maybe_referral) {
+                    in.pushKV("alias", maybe_referral->GetAlias());
+                }
+            } else {
+                debug("could not fetch spent info");
+            }
+
+            vin.push_back(in);
+        }
+
+        if (!isSender) {
+            auto firstInput = vin[0];
+            vin.setArray(); // Clear obj & set type to array
+            vin.push_back(firstInput); // Add only first input
+        }
+
+        entry.push_back(Pair("inputs", vin));
+    }
+
+    int64_t totalAmount = 0;
+    UniValue vout(UniValue::VARR);
+    bool couldBeGrowthReward = false;
+    bool isMarket = false;
+    bool isPoolReward = false;
+
+    for (unsigned int i = 0; i < tx.vout.size(); i++) {
+        const auto& txout = tx.vout[i];
+        UniValue out(UniValue::VOBJ);
+        txnouttype type;
+        std::vector<CTxDestination> addresses;
+        int required;
+        bool isChangeOutput;
+
+        if (ExtractDestinations(txout.scriptPubKey, type, addresses, required)) {
+            const CTxDestination dest = addresses[0];
+            const std::string stringDest = CMeritAddress{dest}.ToString();
+
+            if (!isSender && stringDest != walletAddress) {
+                continue;
+            } else if (isSender && stringDest == walletAddress) {
+                isChangeOutput = true;
+            }
+
+            out.push_back(Pair("address", stringDest));
+
+            uint160 address;
+            if (GetUint160(dest, address)) {
+                const auto maybe_referral = prefviewcache->GetReferral(address);
+                if (maybe_referral) {
+                    out.push_back(Pair("alias", maybe_referral->alias));
+                }
+            }
+        } else {
+            if (type == TX_NULL_DATA && !(isMarket || isPoolReward || tx.IsCoinBase())) {
+                std::string data;
+                hex2ascii(HexStr(txout.scriptPubKey.begin(), txout.scriptPubKey.end()), data);
```

Comment:
What is happening here? You are converting to hex and then converting back? HexStr converts to hex.

Reply:
std::string data(txout.scriptPubKey.begin(), txout.scriptPubKey.end());